native/core/src/execution/datafusion/planner.rs

```diff
@@ -541,6 +542,24 @@ impl PhysicalPlanner {
                         Some(self.create_expr(case_when.else_expr.as_ref().unwrap(), input_schema)?)
                     }
                 };
+
+                // optimized path for CASE WHEN predicate THEN expr ELSE null END
+                if else_phy_expr.is_none() && when_then_pairs.len() == 1 {
```

Comment:
Couldn't we have added this as an optimization in DF itself?

Reply:
I filed https://github.com/apache/datafusion/issues/11484 to track up streaming this